backend/core/src/main/java/org/sonarsource/sonarlint/core/ai/ide/AiHookService.java

```diff
@@ -0,0 +1,126 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.ai.ide;
+
+import jakarta.inject.Inject;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.embedded.server.EmbeddedServer;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.ai.AiAgent;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.ai.GetHookScriptContentResponse;
+
+public class AiHookService {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private static final String BASH_HOOK = "sonarqube_analysis_hook.sh";
+  private static final String PYTHON_HOOK = "sonarqube_analysis_hook.py";
+  private static final String JS_HOOK = "sonarqube_analysis_hook.js";
+
+  private static final String WINDSURF_HOOK_CONFIG = """
+      {
+        "hooks": {
+          "post_write_code": [
+            {
+              "command": "{{SCRIPT_PATH}}",
+              "show_output": false
+            }
+          ]
+        }
+      }
+      """;
+
+  private final EmbeddedServer embeddedServer;
+  private final ExecutableLocator executableLocator;
+
+  @Inject
+  public AiHookService(EmbeddedServer embeddedServer) {
+    this(embeddedServer, new ExecutableLocator());
+  }
+
+  // For testing
+  AiHookService(EmbeddedServer embeddedServer, ExecutableLocator executableLocator) {
+    this.embeddedServer = embeddedServer;
+    this.executableLocator = executableLocator;
+  }
+
+  public GetHookScriptContentResponse getHookScriptContent(AiAgent agent) {
+    var port = embeddedServer.getPort();
+    if (port <= 0) {
+      throw new IllegalStateException("Embedded server is not started. Cannot generate hook script.");
+    }
+
+    var executableType = executableLocator.detectBestExecutable()
+      .orElseThrow(() -> new IllegalStateException("No suitable executable found for hook script generation. " +
+        "Please ensure Node.js, Python, or Bash is available on your system."));
+
+    var scriptContent = switch (executableType) {
+      case NODEJS -> loadTemplateAndReplacePlaceholders(JS_HOOK, port, agent);
+      case PYTHON -> loadTemplateAndReplacePlaceholders(PYTHON_HOOK, port, agent);
+      case BASH -> loadTemplateAndReplacePlaceholders(BASH_HOOK, port, agent);
+    };
+
+    var configContent = generateHookConfiguration(agent);
+    var configFileName = getConfigFileName(agent);
+
+    return new GetHookScriptContentResponse(scriptContent, executableType.getFileName(), configContent, configFileName);
```

Comment:
I think that the name of the enum `ExecutableType` is too generic, and having it contain a very specific hook name is a bit strange. Maybe we can either have another switch or helper method for the `scriptFileName`, or rename the enum to be hook-specific, like `AgentHookExecutableType` 🤔